### ISSUE TITLE
 Admob Java Singleton not found

### ISSUE THREAD

**Issue by tx350z** [Shin-NiL/Godot-Android-Admob-Plugin#24]
**Godot version: v3.2.stable.official

**Issue description:**
Engine.has_singleton("AdMob") returns false.

After spending hours searching and experimenting, I've given up trying to fix this and hope by creating an issue someone will post an answer, update the documentation, or whatever.

I've tried following the documentation, tutorials, examples, suggestions, and even the downloaded the demo project with no success.

I'm guessing (hoping anyway) that the problem is just a missing piece of information for install/setup/use.

Thank you in advance.

**Comment by anisc** [comment]
are you debugging on your phone or computer ?

**Comment by tx350z** [comment]
I am debugging in the editor. Does the plugin work in the editor?

**Comment by Shin-NiL** [comment]
The plugin is Android only, you have to test it on an Android device, it will not work on editor.

**Comment by tx350z** [comment]
Thanks, that is the missing piece. I'll give it a try in the next day or so.

Thank you for the great work!

**Comment by tx350z** [comment]
Tried running the demo on my Moto Z Play. Got basically the same error - could not find singleton for class: org/godotengine/godot/GodotAdMob.

Log cat:
2020-03-23 10:15:39.676 7537-7769/? I/godot: Loading Android module: org/godotengine/godot/GodotAdMob
2020-03-23 10:15:39.677 7537-7769/? E/godot: **ERROR**: Couldn't find singleton for class: org/godotengine/godot/GodotAdMob.
2020-03-23 10:15:39.677 7537-7769/? E/godot:    At: platform/android/java_godot_lib_jni.cpp:691:_initialize_java_modules() - Condition "!singletonClass" is true. Continuing.
    
    
    --------- beginning of crash
2020-03-23 10:15:39.680 7537-7769/? E/AndroidRuntime: FATAL EXCEPTION: GLThread 5367
    Process: org.godotengine.admobplugintest, PID: 7537
    java.lang.ClassNotFoundException: Didn't find class "org/godotengine/godot/GodotAdMob" on path: DexPathList[[zip file "/data/app/org.godotengine.admobplugintest-GldQOe7kb4dJAqHFBYThjA==/base.apk"],nativeLibraryDirectories=[/data/app/org.godotengine.admobplugintest-GldQOe7kb4dJAqHFBYThjA==/lib/arm, /data/app/org.godotengine.admobplugintest-GldQOe7kb4dJAqHFBYThjA==/base.apk!/lib/armeabi-v7a, /system/lib, /system/vendor/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:93)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at org.godotengine.godot.GodotLib.setup(Native Method)
        at org.godotengine.godot.Godot$2.run(Godot.java:325)
        at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1500)
        at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1270)

**Comment by Shin-NiL** [comment]
Did you turn on the "Use Custom Build" option on the Android export window?

**Comment by tx350z** [comment]
Yes, it is on. I decided to try building a simple app using Android studio following an AdMob tutorial. I did eventually get it to work. The problems I ran into getting AdMob working gives me some clues where to look in my Godot app.

I truly believe the problem is mine and is probably caused by some configuration goof. I hope to have more info when I have time to chase it. Will post an update here when I have something.

Thanks.
